sdk/keyvault/keyvault-keys/test/recoverBackupRestore.test.ts

```diff
@@ -33,15 +33,17 @@ describe("Keys client - restore keys and recover backups", () => {
   it("can recover a deleted key", async function() {
     const keyName = testClient.formatName(`${keyPrefix}-${this!.test!.title}-${keySuffix}`);
     await client.createKey(keyName, "RSA");
-    await client.deleteKey(keyName);
-    const getDeletedResult = await retry(async () => client.getDeletedKey(keyName));
+    const deletePoller = await client.beginDeleteKey(keyName);
+    await deletePoller.pollUntilDone();
+    const getDeletedResult = await client.getDeletedKey(keyName);
```

Comment:
Per comment above, might want to test that this works prior to your second `poll()` call (line 37) - at least, that's how we do it in .NET and Python. The idea is that a customer may not have need to wait, and the server already provided everything they need otherwise.

Reply:
done!